File: 프로그래머스/1/1845. 폰켓몬/폰켓몬.cpp
#include <vector>
#include <algorithm>
using namespace std;

int solution(vector<int> nums)
{
    int answer = 0;
    vector<int> temp;
    int size = nums.size()/2;

    sort(nums.begin(), nums.end());
    
    for(int i = 0; i<nums.size();i++){
        temp.push_back(nums[i]);
    }
    temp.erase(unique(temp.begin(), temp.end()), temp.end());
        
    if(temp.size()<size) answer = temp.size();
    else answer = size;
    
    return answer;
}

File: 프로그래머스/1/1845. 폰켓몬/폰켓몬.py
def solution(nums):
    answer = 0
    pocket = {}
    target = len(nums) / 2
    
    for v in nums:
        pocket[v] = 0
        
    if len(pocket) < target:
        answer = len(pocket)
    else:
        answer = target
    
    return answer